src/EffectBase.h

```diff
@@ -73,6 +76,12 @@ namespace openshot
 
 	public:
 
+		/// Parent effect (which properties will set this effect properties)
+		EffectBase* parentEffect;
+
+		/// Map of Tracked Object's by their indices (used by Effects that track objects on clips)
+		std::map<int, std::shared_ptr<TrackedObjectBase> > trackedObjects;
```

Comment:
Something feels off about this, when something so specific to one type of effect has to be a member of the _base_ class for all effects.

Couldn't there be a class that manages this data for effects that do tracking, so that only a pointer or reference to _that_ object is stored in the `EffectBase`?

I dunno, I'm just trying to understand the object hierarchy here, and failing. Especially when you include `TrackedObjectBase` and its **one** derived class. (Why does it need a base class, if there's only one class that derives from it? And if `AddBox` and `GetBoxValues` are present in `TrackedObjectBase`, then the bounding box has been made part of the base class in all but implementation, so the artificial base/derived relationship between `TrackedObjectBase` and `TrackedObjectBBox` feels like a lie, or at a minimum wishful thinking.)